### PR TITLE
Login page refactoring

### LINKED ISSUE #11257: Render Login Directly on Home
##### **Overview of the feature request**

move login inside of home component instead of modal dialog

- less popups
- simpler design
- less bugs?
- in angular

1.  get rid of `login-modal.service.ts`
2.  less dependencies of `loginModalService `in `auth-expired.interceptor`, `homeComponent`, `userRouteAcessService`
3. no more `this.activeModal.dismiss('cancel');`

##### **Related issues or PR**

-   [x] Checking this box is mandatory (this is just to show you read everything)

### PR DESCRIPTION
**Pull request by maznag**
Proposed PR to Fix #11257 and #10733

-   Please make sure the below checklist is followed for Pull Requests.

-   [x] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
-   [x] Tests are added where necessary
-   [ ] Documentation is added/updated where necessary
-   [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

**Comment by avdev4j** [comment]
Hi @maznag 
It seems you've done a great job here!
I suggest to waiting for the v7 branch to merge into (I'm sure this is why @pascalgrimaud put it "on hold"). 

But if we accept this kind of modification, we should do the same for the React part since we offer both React and Angular options. For Vuejs it's not a blocking condition because it will be a feature to migrate into the blueprint.

**Comment by guerreroje** [comment]
Does not navigate to the previous stored url post successful authentication (AccountService.navigateToStoredUrl).

**Comment by maznag** [comment]
> I think there is no need to use StateStorageService. It can be done in this simple way:
> 
> ```js
> if (!this.router.getCurrentNavigation()) {
>    // There were no routing during login (eg from navigationToStoredUrl)
>    this.router.navigate(['']);
> }
> ```

I have not tested your code but it works well with StateStorageService.  With StateStorageService we are sure to navigate to previous URL. Have you tested that I proposed?

**Comment by guerreroje** [comment]
> > I think there is no need to use StateStorageService. It can be done in this simple way:
> > ```js
> > if (!this.router.getCurrentNavigation()) {
> >    // There were no routing during login (eg from navigationToStoredUrl)
> >    this.router.navigate(['']);
> > }
> > ```
> 
> I have not tested your code but it works well with StateStorageService. With StateStorageService we are sure to navigate to previous URL. Have you tested that I proposed?

Yes, I tried it and it works fine. I was just making the proposal to have one less dependency and be able to respond to any routing done during login flow without relying on StateStorageService, eg: route to a custom user home page.

**Comment by maznag** [comment]
> > > I think there is no need to use StateStorageService. It can be done in this simple way:
> > > ```js
> > > if (!this.router.getCurrentNavigation()) {
> > >    // There were no routing during login (eg from navigationToStoredUrl)
> > >    this.router.navigate(['']);
> > > }
> > > ```
> > 
> > 
> > I have not tested your code but it works well with StateStorageService. With StateStorageService we are sure to navigate to previous URL. Have you tested that I proposed?
> 
> Yes, I tried it and it works fine. I was just making the proposal to have one less dependency and be able to respond to any routing done during login flow without relying on StateStorageService, eg: route to a custom user home page.

OKAY ! I'll try to test your proposal

File: generators/client/templates/angular/src/main/webapp/app/login/login.component.ts.ejs
Comment (qmonmert) [inline generators/client/templates/angular/src/main/webapp/app/login/login.component.ts.ejs:79]: parenthesis are really necessary ?

Comment (maznag) [reply generators/client/templates/angular/src/main/webapp/app/login/login.component.ts.ejs:79]: no ! just forgot

File: generators/cleanup.js
Comment (mshima) [inline generators/cleanup.js:59]: Revert this change.
And add removed modal files to jhipster 7 cleanup.

**Review by mshima — CHANGES_REQUESTED** [review]
Drop yarn log.

**Comment by mshima** [comment]
@pascalgrimaud can we go ahead now?

**Review by Yoshio123 — APPROVED** [review]

File: generators/client/templates/angular/src/main/webapp/app/shared/shared.module.ts.ejs
Comment (mshima) [inline generators/client/templates/angular/src/main/webapp/app/shared/shared.module.ts.ejs:51]:
```suggestion
    ]
})
```

**Comment by pascalgrimaud** [comment]
@mshima : yes, sure. I suggested to wait v7. Now, it's ok

**Comment by mshima** [comment]
@maznag CI must pass to be merged.

**Comment by DanielFran** [comment]
@maznag Can you fix the conflicts please?
As soon as this is green we can merge it...

File: generators/client/templates/angular/src/test/javascript/spec/app/login/login.component.spec.ts.ejs
Comment (kaidohallik) [inline generators/client/templates/angular/src/test/javascript/spec/app/login/login.component.spec.ts.ejs:76]: This is called in `comp.ngOnInint();` and this test should test that, so remove this line.

File: generators/client/templates/angular/src/test/javascript/spec/app/login/login.component.spec.ts.ejs
Comment (kaidohallik) [inline generators/client/templates/angular/src/test/javascript/spec/app/login/login.component.spec.ts.ejs:89]:
```suggestion
```
This can be removed after `isAuthenticated` is removed from component.

File: generators/client/templates/angular/src/test/javascript/spec/app/login/login.component.spec.ts.ejs
Comment (kaidohallik) [inline generators/client/templates/angular/src/test/javascript/spec/app/login/login.component.spec.ts.ejs:92]: After `isAuthenticated` is removed from component also this needs adjusting:
```suggestion
      mockAccountService.isAuthenticated = () => true;
```

**Review by kaidohallik — CHANGES_REQUESTED** [review]
Great work @maznag ! All worked as excpected.  
I have some comments about code cleaning and hints how to get tests pass.

File: generators/client/templates/angular/src/main/webapp/app/login/login.component.ts.ejs
Comment (kaidohallik) [inline generators/client/templates/angular/src/main/webapp/app/login/login.component.ts.ejs:92]: One more note. As `accountService.identity()` returns `Observable` from `HttpClient` then there is no need to `unsubscribe` because `HttpClient` completes automatically after emitting value. `unsubscribe` doesn't harm but currently we are not unsubscribing from `HttpClient` observables. To keep this style can you remove unsubscription and related variables?
```suggestion
```

**Comment by maznag** [comment]
Thank you @LL for your help.  the PR is ready for merge.

**Comment by DanielFran** [comment]
@maznag Thanks for thi sawesome work!

I think @kaidohallik already test it, so it should be OK.

@pascalgrimaud Is it OK for you?

**Comment by pascalgrimaud** [comment]
I'm ok too

**Comment by pascalgrimaud** [comment]
@maznag : there is a bounty on each ticket. Don't forget to claim it and mention me :)

**Comment by maznag** [comment]
@pascalgrimaud @DanielFran  thanks a lot. This is the link to OpenCollective expense https://opencollective.com/generator-jhipster/expenses/27271

**Comment by avdev4j** [comment]
Congrats @maznag and thanks for you contribution to the v7 😉

**Comment by pascalgrimaud** [comment]
@maznag : approved

**Pull request merged**
